### DISCUSSION TITLE
GoCardless Security Notification

### DISCUSSION THREAD

**Discussion by digitlength** [firefly-iii/firefly-iii#8865] [Firefly III]
Today I received the below email from GoCardless. I run the Firefly III Importer every hour, which connects to my banks via GoCardless. Does Firefly III use the endpoints called out below during this scenario? If not, does it use those endpoints at any other time?

> Dear GoCardless bank account data client. 
>
>We are writing to inform you that an error occurred in our request caching on the GoCardless Bank Account Data API and Portal between 13:15-14:30 UTC on 25 April 2024. There is a low risk of any customer data having being compromised but we would like to share details of what happened, and the steps that we have taken to address this issue:
>
>Risk of data vulnerability
>Caching is a software component that stores data so that future requests for that data can be served faster. An internal error in the set up of https request caching means that your users' data was vulnerable to other GoCardless Bank Account Data users for a 75 minute period. The data includes IBANs, names of account holders, account details, balances, and transaction history. 
>
>For someone to have accessed data, they would have needed the UUID (account_id). This can only be obtained from /requisitions/<requisition_id>/ or /requisitions/ endpoints.
>
>If you did not use the /requisitions/<requisition_id>/ or /requisitions/ endpoint between 13:15 - 14:30 UTC the likelihood of another user obtaining the ID's necessary to query and access your resources is extremely low. We also confirm that there have been no attempts to scan or brute force the UUIDs in the timeframe of vulnerability.
>
>Actions taken by GoCardless
>As soon as the issue was detected, we turned off the caching putting an immediate stop to generating new caches and the returned caches dropped off in the following minutes.
>
>We immediately launched an internal investigation and have since reported it to the Privacy Regulator in France (CNIL) following all of the necessary protocols.
>
>In the unlikely event that a user’s data was accessed:
>
>IBAN information could be used to set up a Direct Debit. In this instance there is protection in place under the Direct Debit Guarantee or scheme equivalent. 
>
>Transaction history may expose a partially masked card number, including the first 6 and last 4 digits. This information, along with personal details such as full name and address, could be tested and used for unauthorised card payments.  
Unauthorised credit checks could be made against their name and address details
>
>We recommend that you advise your customers to check their accounts and flag any suspicious activity to their bank.
>
>If your customers recognise any unauthorised activity through the GoCardless platform and they would like us to add any restrictions to the bank account used, please feel free to share the email address [bank-account-data-support@gocardless.comhelp@gocardless.com](mailto:bank-account-data-support@gocardless.comhelp@gocardless.com) with them and we will ensure bank accounts are restricted and credit checks are removed. 
>
>Sincerely,
>GoCardless Bank Account Data Team

**Comment by Xeravax** [comment]
From what I am guessing, firefly **might?** use these endpoints, but as far as the UUID's go, these are very hard to obtain/guess. They're random 25 character combinations (if I'm not mistaken) and this is only a formality as they're required to alert you of any **_potential_** breaches. 

tl;dr, unless they had the uuid. They coudn't lookup this data.

**Comment by JC5** [comment]
Thanks for letting me know, Codixer.

I will check out the source to see if the data importer uses those endpoints. I believe so, but I can't be certain.

Either way: if the data importer downloaded somebody else's data I believe it would still fail the import, because it would not be able to match account numbers. 

Perhaps some residual information from other Nordigen accounts would be in your logs. Those would expire in 5 days.
